core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/AddChildProfileViewModel.kt

```diff
@@ -0,0 +1,235 @@
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
+import com.ustadmobile.core.util.ext.appendSelectedAccount
+import com.ustadmobile.core.util.ext.onActiveEndpoint
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.core.viewmodel.UstadEditViewModel
+import com.ustadmobile.core.viewmodel.clazz.list.ClazzListViewModel
+import com.ustadmobile.door.util.systemTimeInMillis
+import com.ustadmobile.lib.db.entities.Person
+import com.ustadmobile.lib.db.entities.PersonParentJoin
+import io.github.aakira.napier.Napier
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.builtins.ListSerializer
+import org.kodein.di.DI
+import org.kodein.di.instance
+
+
+data class AddChildProfileUiState(
+    val onAddChildProfile: String? = null,
+    val childProfiles: List<Person> = emptyList(),
+    val personParenJoinList: List<PersonParentJoin> = emptyList(),
+    val profileIncludingParentAndChildren: List<Person> = emptyList(),
+    val showProfileSelectionDialog: Boolean = false
+)
+
+class AddChildProfileViewModel(
+    di: DI,
+    savedStateHandle: UstadSavedStateHandle,
+) : UstadEditViewModel(di, savedStateHandle, DEST_NAME) {
+
+    private val _uiState = MutableStateFlow(
+        AddChildProfileUiState()
+    )
+    private var nextDestination: String = savedStateHandle[UstadView.ARG_NEXT] ?: ClazzListViewModel.DEST_NAME_HOME
+     val repo: UmAppDatabase by di.onActiveEndpoint().instance()
+
+    val uiState: Flow<AddChildProfileUiState> = _uiState.asStateFlow()
+
+    private val parentToList: MutableList<Person> = mutableListOf(accountManager.currentUserSession.person)
+
+    init {
+        accountManager.activeLearningSpace.url
+
+        _appUiState.update { prev ->
+            prev.copy(
+                title = systemImpl.getString(MR.strings.add_child_profiles),
+                hideBottomNavigation = true,
+            )
+        }
+
+        _appUiState.update { prev ->
+            prev.copy(
+                actionBarButtonState = ActionBarButtonUiState(
+                    visible = true,
+                    text = systemImpl.getString(MR.strings.finish),
+                    onClick = this@AddChildProfileViewModel::onClickFinish
+                )
+            )
+        }
+
+        launchIfHasPermission(
+            permissionCheck = {
+                true
+            }
+        ) {
+            async {
+                loadEntity(
+                    serializer = ListSerializer(Person.serializer()),
+                    loadFromStateKeys = listOf(STATE_KEY_PERSONS),
+                    onLoadFromDb = { db ->
+                        val ChildProfilesDb = db.personDao()
```

Comment:
I don't think we need to load this from the database. In this screen the initial list is always empty.